crates/holochain_zome_types/src/link.rs

```diff
@@ -3,6 +3,38 @@ use crate::ChainTopOrdering;
 use holo_hash::HeaderHash;
 use holochain_serialized_bytes::prelude::*;
 
+#[derive(
+    Debug,
+    Copy,
+    Clone,
+    Hash,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    serde::Serialize,
+    serde::Deserialize,
+    SerializedBytes,
+)]
+#[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
+pub struct LinkType(pub u8);
+
+impl LinkType {
+    pub fn new(u: u8) -> Self {
+        Self(u)
+    }
+
+    pub fn into_inner(self) -> u8 {
+        self.0
+    }
+}
+
+impl From<u8> for LinkType {
+    fn from(u: u8) -> Self {
+        Self(u)
+    }
+}
```

Comment:
Hmm is this actually what we want? Seems like it encourages devs to use plain u8 instead of defining their own LinkType. They could use `new` if they want to use a bare byte. I'm in favor not including this From impl.